### ISSUE TITLE
SVM destination native warp routes should be rent-aware

### ISSUE THREAD

**Issue by tkporter** [hyperlane-xyz/hyperlane-warp-ui-template#340]
Context https://discord.com/channels/935678348330434570/1264883412188135555/1311371339889639486

Copying some of the important bits here:

there is a single SOL transfer eclipse -> solana that cannot be processed

Jito says that it's invalid, which I believe means it'll revert

the message
```
"PendingMessage { num_retries: 20, since_last_attempt_s: 1118, next_attempt_after_s: 0, message: HyperlaneMessage { id: 0xe955d02bfac7dd239a290cfa0319ff73f59517304f5e93a85939ddb6c52aadce, version: 3, nonce: 50011, origin: eclipsemainnet, sender: 0xd49959a478c4c4170ef86528b8e14646d0a94b4ecbc0582f28d7e4de2e4f5b44, destination: solanamainnet, recipient: 0x6b4e9fed101f020c0b989737843efb49ac1aa918aa992c4a3808d29bc12f7efd, body: 0x64a91dfb87f287d85ec1b2dce42dce1a749e948a2005cce647ee8fceac67b7aa00000000000000000000000000000000000000000000000000000000000c3500 }, status: ReadyToSubmit, app_context: Some("SOL/eclipsemainnet-solanamainnet") }"
```

it's of a very small amount of SOL
```
$ cast td 0xc3500
800000
```

Which is `0.0008 SOL`
and the recipient has no balance
which makes me think this is maybe a rent thing
```
$ solana rent 0 -u mainnet-beta
Rent-exempt minimum: 0.00089088 SOL
```

so the only way this goes through is if they have at least 0.00089088 after the transfer

**TLDR:**
we should enforce on SVM destinations for native warp routes (collateral / synthetic warp routes aren't subject to this) in the UI that either:
1. the recipient is already rent exempt
2. the native token amount bridged will exceed rent exemption
